components/Joke.js

```diff
@@ -0,0 +1,100 @@
+import React from "react";
+import { useState, useEffect } from "react";
+import styled from "styled-components/native";
+import { Vibration } from "react-native";
+
+const Joke = ({ route }) => {
+  const [joke, setJoke] = useState({});
+  const { itemId } = route.params;
+
+  const JOKE_URL = `https://sv443.net/jokeapi/v2/joke/${itemId}`;
+
+  useEffect(() => {
+    reFetchData();
+  }, []);
+
+  const reFetchData = () => {
+    fetch(JOKE_URL)
+      .then(response => response.json())
+      .then(json => {
+        if (json.flags.racist || json.flags.religious || json.flags.nsfw) {
+          reFetchData();
```

Comment:
Great job, I remember our 1:1 session regarding this topic!
If you would like to understand what's happening over here a bit deeper, take a look on this article : https://javascript.info/recursion